### PR TITLE
test: Filter docker integration tests for Firecracker

### LINKED ISSUE #1104: Filter docker integration tests for Firecracker
Currently not all the docker integration tests are running with Firecracker so with this change we filter the tests.

### PR DESCRIPTION
**Pull request by GabyCT**
Currently not all the docker integration tests are running
with Firecracker so with this change we filter the tests.

Fixes #1104

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: .ci/hypervisors/firecracker/filter_docker_firecracker.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		_skip_options+=("${entry#- }|")
+	done
+}
+
+main() {
+	# Check GOPATH is set
+	check_gopath
+
+	# Check if yq is installed
+	[ -z "$(command -v yq)" ] && install_yq
```
[the remark this answers is not in the document]
Comment (GabyCT) [reply .ci/hypervisors/firecracker/filter_docker_firecracker.sh]: @grahamwhaley `check_cmds` fails if it is not present and that is only being used in `metrics`.

File: .ci/hypervisors/firecracker/configuration_firecracker.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# We need to skip some docker integration tests as they are not
+# running correctly using Firecracker. We will skip them using
+# the ginkgo flag 'skip=REGEXP'.
+test:
```
[the remark this answers is not in the document]
Comment (GabyCT) [reply .ci/hypervisors/firecracker/configuration_firecracker.yaml]: @grahamwhaley no there is not.

File: Makefile [hunk 3]
```diff
@@ -12,6 +12,9 @@ UNION := functional docker crio docker-compose network netmon docker-stability o
 
 # skipped test suites for docker integration tests
 SKIP :=
+ifeq ($(KATA_HYPERVISOR), firecracker)
+	SKIP := $(shell bash -f .ci/hypervisors/firecracker/filter_docker_firecracker.sh)
```
[the remark this answers is not in the document]
Comment (GabyCT) [reply Makefile]: @grahamwhaley , I am not sure that I understand :S, so where are we going to say that if it is `firecracker` do the skip ? (if we are dropping the `ifeq`)

[the remark this answers is not in the document]
Comment (GabyCT) [reply Makefile]: @grahamwhaley changes applied

**Comment by jodh-intel** [comment]
lgtm

File: .ci/hypervisors/firecracker/configuration_firecracker.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (grahamwhaley) [reply .ci/hypervisors/firecracker/configuration_firecracker.yaml]: not what - no docs from Ginkgo or no docs for the format we invented? I think we need to document the meaning and use of `Context` and `It`  and the other identifiers for instance...

**Comment by devimc** [comment]
lgtm

File: Makefile [hunk 4]
```diff
@@ -11,7 +11,7 @@ TIMEOUT := 60
 UNION := functional docker crio docker-compose network netmon docker-stability oci openshift kubernetes swarm vm-factory entropy ramdisk shimv2
 
 # skipped test suites for docker integration tests
-SKIP :=
+SKIP := $(shell bash -f .ci/hypervisors/$(KATA_HYPERVISOR)/filter_docker_$(KATA_HYPERVISOR).sh)
```
Comment (devimc) [inline Makefile]: hold on. what happen when KATA_HYPERVISOR is qemu? :thinking:

Comment (GabyCT) [reply Makefile]: http://jenkins.katacontainers.io/job/kata-containers-tests-ubuntu-16-04-PR/1108/consoleFull

Comment (devimc) [reply Makefile]: thanks @GabyCT

**Pull request merged**